File: task_equivalents/llm_translations/starcoder/optimized/r/data_manipulation_preprocessing/01_impute_encode_scale.R
library(dplyr)
library(tidyr)
library(caret)
Auto <- read.csv("data/csv/Auto.csv", na.strings="?") %>%
  drop_na() %>%
  mutate(mpg01 = ifelse(mpg > median(mpg), 1, 0)) %>%
  select(-mpg, -name)
var_no_scale <- c("cylinders", "year", "origin", "mpg01")
var_to_scale <- c("displacement", "horsepower", "weight", "acceleration")
scaled_var <- scale(Auto[, var_to_scale])
df <- cbind(scaled_var, Auto[, var_no_scale])
x <- df[, -ncol(df)]
y <- df[, ncol(df)]

File: task_equivalents/llm_translations/starcoder/optimized/python/data_manipulation_preprocessing/01_impute_encode_scale.py
import pandas as pd
from sklearn import preprocessing

def translate_code():
    df = pd.read_csv('data/csv/Auto.csv', na_values='?', index_col=1)
    df = df.dropna()

    df['mpg01'] = (df['mpg'] > df['mpg'].median()).astype(int)
    df = df.drop(['mpg', 'name'], axis=1)
    
    var_no_scale = ['cylinders', 'year', 'origin', 'mpg01']
    var_to_scale = ['displacement', 'horsepower', 'weight', 'acceleration']
    scaled_var = preprocessing.scale(df[var_to_scale])
    df = pd.concat([pd.DataFrame(scaled_var, columns=var_to_scale),
                    df[var_no_scale]], axis=1)
    
    x = df.drop('mpg01', axis=1)
    y = df['mpg01']

    return x, y
x, y = translate_code()
